Fix/idea detail logic

goyadev opened the pull request:
- Added EOI to show yes or no
- Fixed Date
- added button logic eg. if no mentors for workshop idea, no button for mentors show. If workshop is closed, EOI buttons do not render, but a contact organiser button renders instead.

![Screenshot 2024-04-04 at 11 48 03 pm](https://github.com/SheCodesAus/2024_exception_al_frontend/assets/143305696/8a9985db-c15d-4d14-b1e9-63e02f52c3d3)
![Screenshot 2024-04-04 at 11 47 49 pm](https://github.com/SheCodesAus/2024_exception_al_frontend/assets/143305696/ee0681d4-d1ae-44d1-8b45-27be3fd0b88e)
![Screenshot 2024-04-04 at 11 44 13 pm](https://github.com/SheCodesAus/2024_exception_al_frontend/assets/143305696/9b612db9-a104-4736-a1fd-7ab001417192)

goyadev committed:
fixed taking new EOI code to show yes or no

goyadev committed:
found the pesky cause of the date error, it was an extra 'n' in 'planned'

goyadev committed:
added logic so that when the workshop has no mentors, the button doesn't render. When the workshop is closed, both EOI buttons don't render, but a contact organiser button renders instead

goyadev committed:
updated logic for if workshop, open or closed, needs mentors or not

goyadev committed:
Merge branch 'development' into Fix/idea-detail-logic

suanTech commented:
Nice! I think the buttons need some spacing between, and also will be nice to have bold text for the details title like Proposed date and location etc,... need a bit of separation from the header part and the description part!

suanTech commented on src/pages/IdeaDetail.jsx:
```diff
@@ -2,31 +2,38 @@ import { useState, useEffect } from "react";
   const [formattedCost, setFormattedCost] = useState("");
+  const [isOpenText, setIsOpenText] = useState("");
 
   useEffect(() => {
```
useEffect here is not needed, it can be a plain function!

suanTech commented on src/pages/IdeaDetail.jsx:
```diff
@@ -2,31 +2,38 @@ import { useState, useEffect } from "react";
+            style: "currency",
+            currency: "AUD",
+          }).format(cost);
+          setFormattedCost(formattedCost);
```
formattedCost can be just a variable, instead of using useState.
can be 
```

const formattedCost = formatCost();
const formatConst = () => {
let formattedCost = "";
if (workshop && workshop.est_cost) {
        const cost = workshop.est_cost;
        formattedCost = new Intl.NumberFormat("en-AU", {
          style: "currency",
          currency: "AUD",
        }).format(cost);
 return formattedCost
}
```

And this way you can use the formatted cost variable where you need.

goyadev replied:
updated code

suanTech commented on src/pages/IdeaDetail.jsx:
```diff
@@ -41,7 +48,7 @@ export default function IdeaDetail() {
           <p>Proposed Date: {formattedDate}</p>
           <p>Location: Perth</p>
-          <p>Taking new EOIs: {workshop.is_open}</p>
+          <p>Taking new EOIs: {isOpenText}</p>
```
it can be {workshop.isOpen ? "Yes" : "No"}

suanTech reviewed (changes requested):
Just a few spots that can improve readability!

goyadev committed:
added spacing and bolding

goyadev committed:
added some margin around main text area

goyadev committed:
updated code around formattedCost

goyadev commented:
-Updated code for formatted cost
- Added styling suggestions with more space between heading and text, buttons, and bolding of some text
![Screenshot 2024-04-05 at 8 51 19 pm](https://github.com/SheCodesAus/2024_exception_al_frontend/assets/143305696/074316e4-39e0-4509-9807-0340c597c2f3)

suanTech reviewed (approved):
niceee
